### PR TITLE
Remove duplication found in tests and unify namespaces of common code

### PR DESCRIPTION
**Pull request by chamons**

### CODE REVIEW

**Comment by xamarin-release-manager** [comment]
Build success

**Comment by xamarin-release-manager** [comment]
Build success

**Comment by xamarin-release-manager** [comment]
Build failure

File: tests/dontlink-mac/Stub.cs [hunk 1]
```diff
@@ -3,5 +3,5 @@
 namespace MonoMacFixtures {
 	[TestFixture]
 	public class StubTest {
-	}
-}
+		// dontlink-mac is currently obsolete due to movement of all tests to introspection tests
+		// But in the future, we'll add tests like iOS that actually test linking here
```
Comment (spouliot) [inline tests/dontlink-mac/Stub.cs]: looks like it removed the ending `}` and broke the build step

**Comment by xamarin-release-manager** [comment]
Build success

**Pull request merged**